plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveWriteUtils.java

```diff
@@ -728,24 +736,34 @@ public static HiveDecimal getHiveDecimal(DecimalType decimalType, Block block, i
         return HiveDecimal.create(unscaledValue, decimalType.getScale());
     }
 
-    private static Timestamp getHiveTimestamp(TimestampType type, Block block, int position)
+    private static Timestamp getHiveTimestamp(DateTimeZone localZone, TimestampType type, Block block, int position)
     {
         verify(type.getPrecision() <= HiveTimestampPrecision.MAX.getPrecision(), "Timestamp precision too high for Hive");
 
-        long epochMicros;
-        int nanosOfMicro;
+        long epochMicro;
+        int nanoOfMicro;
```

Comment:
Keep them plural: epochMicros, nanosOfMicro

(i think we're pretty consistent about 'epochMicros' pluralness, but i know we're pretty inconsisten about `unitsOfUnit` pluralness, but i think we're leaning towards plural form overall)

Reply:
I've changed the `epochUnits` to be plural, but I left the `unitOfSecond`s singular.

I think it's more correct, because I read `nanoOfSecond = 3` as "the third nanosecond of the second," not something like "three nanoseconds into the second"—basically the same reason I would use `minuteOfHour` and `hourOfDay` over `minutesOfHour` and `hoursOfDay`.

We're only somewhat inconsistent with the `OfSecond` forms—we lean slightly towards the plural forms in most cases, except for picoseconds,  where we lean very, very heavily towards the plural: `picosOfMilli` and `picosOfMicro` alone are used (as variables) more than every other `unitOfSecond` and `unitsOfSecond` form combined; without those two, `___OfSecond` and `___sOfSecond` are used almost equally as variable names).

Reply:
see io.trino.spi.type.LongTimestamp and io.trino.spi.type.LongTimestampWithTimeZone -- both have `picosOfUnit`.
my intent was to stay consistent with them

Reply:
I'll change it.

I think some places should still use `nanoOfSecond` (e.g. the methods that use `minuteOfHour`), but we do use `getPicosOfMicro` here, so `nanosOfSecond` is consistent even just within this method.